public/app/features/alerting/unified/components/rule-editor/ConditionField.tsx

```diff
@@ -31,10 +32,13 @@ export const ConditionField: FC = () => {
   // reset condition if option no longer exists or if it is unset, but there are options available
   useEffect(() => {
     const expressions = queries.filter((query) => query.datasourceUid === ExpressionDatasourceUID);
-    if (condition && !options.find(({ value }) => value === condition)) {
-      setValue('condition', expressions.length ? expressions[expressions.length - 1].refId : null);
-    } else if (!condition && expressions.length) {
-      setValue('condition', expressions[expressions.length - 1].refId);
+    const lastExpression = last(expressions);
+    const conditionExists = options.find(({ value }) => value === condition);
+
+    if (condition && !conditionExists) {
+      setValue('condition', lastExpression?.refId ?? null);
+    } else if (!condition && lastExpression) {
+      setValue('condition', lastExpression.refId, { shouldValidate: true });
```

Comment:
This fixes a subtle bug where you would not be able to preview anything if you haven't changed any queries or manually selected a condition from the dropdown